core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -103,34 +122,45 @@ public void close() throws IOException {
      */
     public void abort() throws IOException {
         close();
-        tmpFile.delete();
+        Files.deleteIfExists(tmpFile);
     }
 
     public void commit() throws IOException {
         close();
-        if (destFile.exists()) {
-            try {
-                Util.deleteFile(destFile);
-            } catch (IOException x) {
-                tmpFile.delete();
-                throw x;
-            }
+        try {
+            // Try to make an atomic move.
+            Files.move(tmpFile, destFile, StandardCopyOption.ATOMIC_MOVE);
```

Comment:
Do you not _also_ want `REPLACE_EXISTING` here? In case `destFile` already existed? (Is this case tested?)

Reply:
>  implementation specific 
So though in practice @jtnord is right, I would agree with @jglick we should use `REPLACE_EXISTING` too.

Because, though in my testing on my machine, it does not make indeed any difference. I think this better conveys the thing we are willing to do. And it seems not putting `REPLACE_EXISTING` could fail on a implementation and **we** would be at fault.

Reply:
it is pointless - it is ignored even if that is what you are attempting to do - it means the person that next reads this code goes erm why are they doing X below and the javadoc says its ignored....
(basically atomic-move is  most likley a posix `rename()` which will overwrite and no other options make any sense.  however this is implementation specific and a provided could pass both down and an OS implementation coudl retrurn an error with "incomatable_flags"